py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,252 @@
+                    except:
+                        data['netgen'] = "?"
+
+                    # if registred on network, get operator name
+                    try:
+                        if self.rstates[status['m3gpp-registration-state']] in ('HOME', 'ROAMING'):
+                            data['operator'] = status['m3gpp-operator-name']
+                        else:
+                            data['operator'] = STRING_UNKNOWN
+                    except:
+                        data['operator'] = STRING_UNKNOWN
+
+                    # use human readable format
+                    data['status'] = self.states[status['state']]
+
+                    if status['state'] == 11:  # connected
+                        # Get network config
+                        bearer = modem_proxy.Bearers[0]
+                        network_config = self._get_network_config(bearer)
+
+                        # Add network config to data dict
+                        data.update(network_config)
+
+                        if data['ip']:
+                            if self.consider_3G_degraded and data['netgen'] != 'LTE':
+                                response['color'] = self.py2.COLOR_DEGRADED
+                            else:
+                                response['color'] = self.py3.COLOR_GOOD
+                            response['full_text'] = self.py3.safe_format(
+                                self.format_up, data)
+                        else:
+                            response['color'] = self.py3.COLOR_DEGRADED
+                            response['full_text'] = self.py3.safe_format(
+                                self.format_no_service, data)
+
+                    else:  # disconnected
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_down, data)
+                        response['color'] = self.py3.COLOR_BAD
+
+                except:
+                    data['error'] = self.states[status['state']]
+                    response['color'] = self.py3.COLOR_BAD
+                    response['full_text'] = self.py3.safe_format(self.format_error, data)
+
+        return response
+
+    # get network config function
+    def _get_network_config(self, bearer):
+        try:
+            bearer_proxy = self.bus.get(STRING_MODEMMANAGER_DBUS, bearer)
+
+            network_config = {}
+
+            # Get ipv4 config
+            ipv4 = bearer_proxy.Ip4Config
+            if ipv4['address']:
+                network_config['ipv4_address'] = ipv4['address']
+                network_config['ipv4_dns1'] = ipv4['dns1']
+                network_config['ipv4_dns2'] = ipv4['dns2']
+                network_config['ip'] = ipv4['address']
+            else:
+                network_config['ipv4_address'] = ''
+                network_config['ipv4_dns1'] = ''
+                network_config['ipv4_dns2'] = ''
+                network_config['ip'] = STRING_NO_IP
```

Comment:
We shouldn't be nice here with all those `''`. What we want to do first is to simplify the code as much as possible and put everything in the `format`. This seems like a type where we might want to do `format = '{format_ip4} {format_ip6}'` and then use `format_ip4 = '{address} {dns1} {dns2}'`. Same for `format_ip6`.. then users can decide if they want IP4 and/or IP6. We can have both in default `format` in case there are nothing in IP6... and it would stay hidden.